File: Count_All_Valid_Pickup_and_Delivery_Options.cpp
class Solution {
public:
    int countOrders(int n) {
        const int MOD = 1000000007;
        long long totalWays = 1;
        for (int orderNumber = 2; orderNumber <= n; orderNumber++) {
            totalWays = (totalWays * (2LL * orderNumber - 1) * orderNumber) % MOD;
        }
        return static_cast<int>(totalWays);        
    }
};

File: Count_All_Valid_Pickup_and_Delivery_Options.js
/**
 * @param {number} n
 * @return {number}
 */
var countOrders = function(n) {
    const MOD = 1000000007;
    let totalWays = 1;
    for (let orderNumber = 2; orderNumber <= n; orderNumber++) {
        totalWays = (totalWays * (2 * orderNumber - 1) * orderNumber) % MOD;
    }
    return totalWays;    
};
